test/parallel/test-tls-multi-key.js

```diff
@@ -2,52 +2,162 @@
 const common = require('../common');
 const assert = require('assert');
 
+// Test multi-identity ('key')/multi-algorithm scenarios.
+
 if (!common.hasCrypto) {
   common.skip('missing crypto');
   return;
 }
 const tls = require('tls');
 const fs = require('fs');
 
-const options = {
+// Key and cert are ordered as ec, rsa.
```

Comment:
I did not change the test, it pre-exists: https://github.com/nodejs/node/blob/4f638f6fd73c0936cfa088ebb1d5c075c9565b85/test/parallel/test-tls-multi-key.js#L12-L21

Having a mixture of a self-signed cert+key, and cert+key that is issued by a CA is possible, and supported by the API (obviously, because the tests passes), and I think its good to test unusual scenarios as well as usual/recommended.

I agree that there should be tests with the same tier level, thats why this PR exists. It adds tests that use independent cert chains, where the cert chains are all multi-tier (have intermediate certs). The test you commented on below is like that, for example. And in order to write those tests, I first had to add generation of multi-tier certs, because there weren't any for ECC.